test/jdk/java/util/concurrent/ThreadPerTaskExecutor/ThreadPerTaskExecutorTest.java

```diff
@@ -261,17 +253,20 @@ void testClose3(ExecutorService executor) throws Exception {
     @ParameterizedTest
     @MethodSource("executors")
     void testClose4(ExecutorService executor) throws Exception {
-        Future<?> future;
+        Future<Void> future;
         try (executor) {
-            future = executor.submit(SLEEP_FOR_A_DAY);
-            scheduleInterrupt(Thread.currentThread(), Duration.ofMillis(500));
+            var task = new LongRunningTask<Void>();
+            future = executor.submit(task);
+            task.awaitStarted();
+            scheduleInterruptAt("java.util.concurrent.ThreadPerTaskExecutor.close");
         } finally {
             assertTrue(Thread.interrupted());
         }
         assertTrue(executor.isShutdown());
         assertTrue(executor.isTerminated());
-        assertTrue(executor.awaitTermination(10,  TimeUnit.MILLISECONDS));
-        assertThrows(ExecutionException.class, future::get);
+        assertTrue(executor.awaitTermination(10, TimeUnit.MILLISECONDS));
+        Throwable e = assertThrows(ExecutionException.class, future::get);
+        assertTrue(e.getCause() instanceof InterruptedException);
```

Comment:
Would using `assertEquals(InterruptedException.class, e.getCause().getClass())` be better? That way if/when the test fails, it even prints the unexpected exception type?

But I then see that this test already uses `assertTrue` for similar cases in some other place, so maybe it's fine in the current form?

Reply:
I don't mind moving this but it would require changing them everywhere (as you point out).

Reply:
Yes, as we really want an instanceof check here, the exception could be a subclass.